gimie/sources/github.py

```diff
@@ -31,6 +33,24 @@
 from gimie.graph.namespaces import SDO
 
 GH_API = "https://api.github.com"
+load_dotenv()
+
+
+def _set_auth():
+    try:
+        github_token = os.environ.get("ACCESS_TOKEN")
+        assert github_token
+        headers = {"Authorization": f"token {github_token}"}
+
+        login = requests.get("https://api.github.com/user", headers=headers)
+        assert login.json().get("login")
+    except AssertionError:
+        return {}
+    else:
+        return headers
+
+
+GH_HEADERS = _set_auth()
```

Comment:
Doesn't that mean GH_HEADERS will be defined at import time? Wouldn't that be an issue if someone uses `GithubExtractor` in an interactive session?

1. Run `GithubExtractor(...).extract()` → fails due to rate limit
2. export ACCESS_TOKEN=...
3. Reruns `GithubExtractor(...).extract()`  → still fails

Would it make sense to call `_set_auth()` at request time? Or maybe upon instantiation?

Reply:
I agree with @cmdoret, it would make sense to call `_set_auth()` in `extract()` or in `_request()`.

Also, is there a particular reason why `_set_auth()` is a function and not a method of the `GithubExtractor` class? Would it make sense to have a `_set_auth()` method and a `GH_HEADERS` attribute for the class instead?

Reply:
Thanks for these suggestions @cmdoret and @martinfontanet: I made `_set_auth` a class method on the `GithubExtractor`.  I think dictionaries are not allowed as slots of a dataclass, so the header dictionary cannot be an attribute of the `GithubExtractor`. I would only be able to store the token as an attribute. 

Do you think this is better now? Or should I rather take the token approach and make the token a class attribute?